From e7c33f24ed43189c84cc3a31260ad4963a6bb9c7 Mon Sep 17 00:00:00 2001
From: Edward Hervey <edward@centricular.com>
Date: Tue, 11 Jul 2017 15:29:44 +0200
Subject: [PATCH 1/3] pkgconfig: Add private requirements

Add libunwind and dw to the .pc Requires.private. Fixes static library
compilation if gstreamer was compiled with one of those dependencies

https://bugzilla.gnome.org/show_bug.cgi?id=784795
---
 configure.ac                          | 4 ++++
 pkgconfig/gstreamer-uninstalled.pc.in | 2 +-
 pkgconfig/gstreamer.pc.in             | 2 +-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index b6b2923..de9fcae 100644
--- a/configure.ac
+++ b/configure.ac
@@ -824,12 +824,16 @@ dnl libunwind is optionally used by the leaks tracer
 PKG_CHECK_MODULES(UNWIND, libunwind, HAVE_UNWIND=yes, HAVE_UNWIND=no)
 if test "x$HAVE_UNWIND" = "xyes"; then
   AC_DEFINE(HAVE_UNWIND, 1, [libunwind available])
+  UNWIND_REQUIRE=libunwind
+  AC_SUBST(UNWIND_REQUIRE)
 fi
 
 dnl libdw is optionally used to add source lines and numbers to backtraces
 PKG_CHECK_MODULES(DW, libdw, HAVE_DW=yes, HAVE_DW=no)
 if test "x$HAVE_DW" = "xyes"; then
   AC_DEFINE(HAVE_DW, 1, [libdw available])
+  DW_REQUIRE=libdw
+  AC_SUBST(DW_REQUIRE)
 fi
 
 dnl Check for backtrace() from libc
diff --git a/pkgconfig/gstreamer-uninstalled.pc.in b/pkgconfig/gstreamer-uninstalled.pc.in
index b8dcdf8..d071612 100644
--- a/pkgconfig/gstreamer-uninstalled.pc.in
+++ b/pkgconfig/gstreamer-uninstalled.pc.in
@@ -15,6 +15,6 @@ Name: GStreamer Uninstalled
 Description: Streaming media framework, Not Installed
 Version: @VERSION@
 Requires: glib-2.0, gobject-2.0
-Requires.private: gmodule-no-export-2.0
+Requires.private: gmodule-no-export-2.0 @UNWIND_REQUIRE@ @DW_REQUIRE@
 Libs: -L${libdir} -lgstreamer-@GST_API_VERSION@
 Cflags: -I@abs_top_srcdir@ -I@abs_top_srcdir@/libs -I@abs_top_builddir@ -I@abs_top_builddir@/libs
diff --git a/pkgconfig/gstreamer.pc.in b/pkgconfig/gstreamer.pc.in
index b949d6c..6ec3cb7 100644
--- a/pkgconfig/gstreamer.pc.in
+++ b/pkgconfig/gstreamer.pc.in
@@ -13,6 +13,6 @@ Name: GStreamer
 Description: Streaming media framework
 Version: @VERSION@
 Requires: glib-2.0, gobject-2.0
-Requires.private: gmodule-no-export-2.0
+Requires.private: gmodule-no-export-2.0 @UNWIND_REQUIRE@ @DW_REQUIRE@
 Libs: -L${libdir} -lgstreamer-@GST_API_VERSION@
 Cflags: -I${includedir}
-- 
2.10.2


From 1c7f9a5025b7ec0b7cfed701e2a3681a2d2afe74 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Tue, 11 Jul 2017 16:15:16 +0100
Subject: [PATCH 2/3] meson: pkgconfig: add libunwind/libdw to gstreamer-1.0
 Requires.private

https://bugzilla.gnome.org/show_bug.cgi?id=784795
---
 pkgconfig/meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pkgconfig/meson.build b/pkgconfig/meson.build
index 32f96cd..ae27ce3 100644
--- a/pkgconfig/meson.build
+++ b/pkgconfig/meson.build
@@ -7,6 +7,10 @@ pkgconf.set('includedir', '${prefix}/@0@'.format(get_option('includedir')))
 pkgconf.set('GST_API_VERSION', apiversion)
 pkgconf.set('VERSION', gst_version)
 
+# Requires.private
+pkgconf.set('UNWIND_REQUIRE', cdata.has('HAVE_UNWIND') ? 'libunwind' : '')
+pkgconf.set('DW_REQUIRE', cdata.has('HAVE_DW') ? 'libdw' : '')
+
 # needed for generating -uninstalled.pc files
 pkgconf.set('abs_top_builddir', join_paths(meson.current_build_dir(), '..'))
 pkgconf.set('abs_top_srcdir', join_paths(meson.current_source_dir(), '..'))
-- 
2.10.2


From 2d358f8a4ab52574291927c7061109a74e8d12d7 Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Fri, 11 Aug 2017 21:17:06 +0200
Subject: [PATCH 3/3] configure: Add switches for enabling/disabling libdw and
 libunwind

https://bugzilla.gnome.org/show_bug.cgi?id=778193
---
 configure.ac | 46 ++++++++++++++++++++++++++++++++++++----------
 1 file changed, 36 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index de9fcae..6e0019a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -821,19 +821,45 @@ fi
 AM_CONDITIONAL(HAVE_GTK, test "x$HAVE_GTK" = "xyes")
 
 dnl libunwind is optionally used by the leaks tracer
-PKG_CHECK_MODULES(UNWIND, libunwind, HAVE_UNWIND=yes, HAVE_UNWIND=no)
-if test "x$HAVE_UNWIND" = "xyes"; then
-  AC_DEFINE(HAVE_UNWIND, 1, [libunwind available])
-  UNWIND_REQUIRE=libunwind
-  AC_SUBST(UNWIND_REQUIRE)
+AC_ARG_WITH([unwind],[AS_HELP_STRING([--with-unwind=yes|no|auto],[use libunwind])],
+            [], [with_unwind=auto])
+if [ test "x${with_unwind}" != "xno" ]; then
+  PKG_CHECK_MODULES(UNWIND, [libunwind],
+      [
+        HAVE_UNWIND=yes
+        AC_DEFINE(HAVE_UNWIND, 1, [libunwind available])
+        UNWIND_REQUIRE=libunwind
+        AC_SUBST(UNWIND_REQUIRE)
+      ],
+      [
+        HAVE_UNWIND=no
+        if [ test "x${with_unwind}" = "xyes" ]; then
+          AC_MSG_ERROR([could not find libunwind])
+        fi
+      ])
+else
+  HAVE_UNWIND=no
 fi
 
 dnl libdw is optionally used to add source lines and numbers to backtraces
-PKG_CHECK_MODULES(DW, libdw, HAVE_DW=yes, HAVE_DW=no)
-if test "x$HAVE_DW" = "xyes"; then
-  AC_DEFINE(HAVE_DW, 1, [libdw available])
-  DW_REQUIRE=libdw
-  AC_SUBST(DW_REQUIRE)
+AC_ARG_WITH([dw],[AS_HELP_STRING([--with-dw=yes|no|auto],[use libdw])],
+            [], [with_dw=auto])
+if [ test "x${with_dw}" != "xno" ]; then
+  PKG_CHECK_MODULES(DW, [libdw],
+      [
+        HAVE_DW=yes
+        AC_DEFINE(HAVE_DW, 1, [libdw available])
+        DW_REQUIRE=libdw
+        AC_SUBST(DW_REQUIRE)
+      ],
+      [
+        HAVE_DW=no
+        if [ test "x${with_dw}" = "xyes" ]; then
+          AC_MSG_ERROR([could not find libdw])
+        fi
+      ])
+else
+  HAVE_DW=no
 fi
 
 dnl Check for backtrace() from libc
-- 
2.10.2

